pkg/minikube/bootstrapper/bsutil/kubeadm.go

```diff
@@ -74,7 +74,7 @@ func GenerateKubeadmYAML(cc config.ClusterConfig, n config.Node, r cruntime.Mana
 		return nil, errors.Wrap(err, "generating extra component config for kubeadm")
 	}
 
-	cnm, err := cni.New(cc)
```

Comment:
Comment why you have to pass by pointer ? What field do u care about

Reply:
in cni.Net, configureCNI func is called to set the kubelet.cni-conf-dir to '/etc/cni/net.mk' value via cni.ConfDir for docker containerd (crio has better option to define cni network via its 'cni_default_network' config param), so it needs access to cc to set value of KubernetesConfig.ExtraOptions for 'kubelet.cni-conf-dir'